R/examine-constraints.R

```diff
@@ -120,7 +120,7 @@ kind_to_long <- function(kind) {
 
 check_pk_constraints <- function(dm, progress = NA, top_level_fun = NULL) {
   pks <- bind_rows(
-    list(PK = dm_get_all_pks_impl(dm), UK = dm_get_all_uks_impl(dm)),
+    list(PK = dm_get_all_pks_impl(dm), UK = dm_get_all_uks_impl(dm) %>% rename(pk_col = uk_col)),
```

Comment:
There are subtleties with checking a unique key because multiple `NULL` values do not disturb uniqueness (except, of course, in SQL Server 😠 ).

> A unique constraint on T is satisfied if and only if there do not exist two rows R1 and R2 of T such that R1 and R2 have the same non-NULL values in the unique columns.

https://sqlperformance.com/2020/03/t-sql-queries/null-complexities-part-4-missing-standard-unique-constraint is an excellent overview, quote taken from there.

I suggest that we create a `check_uk_constraints()` and (later) optional arguments for `check_key()` that control this.

Reply:
OK, would postpone this to the next PR